ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.js

```diff
@@ -240,35 +240,58 @@
     /**
      * Unassign the role to a contact
      *
-     * @param {string} role role
+     * @param {object} role role
      */
     $scope.unassignRole = function (role) {
-      CRM.confirm({
-        title: ts('Remove %1', { 1: role.role }),
-        message: ts('Remove %1 as %2?', { 1: role.display_name, 2: role.role })
-      }).on('crmConfirm:yes', function () {
-        var apiCalls = [];
-
-        // when client
-        if (!role.relationship_type_id) {
-          apiCalls = [unassignClientCall(role)];
+      // when client
+      if (!role.relationship_type_id) {
+        CRM.confirm({
+          title: ts('Remove %1', { 1: role.role }),
+          message: ts('Remove %1 as %2?', { 1: role.display_name, 2: role.role })
+        }).on('crmConfirm:yes', function () {
+          var apiCalls = [unassignClientCall(role)];
           getApiParamsToSetRelationshipsAsInactiveWhenClientIsRemoved(role, apiCalls);
-        } else {
-          apiCalls = [unassignRoleCall(role)];
-        }
 
-        apiCalls.push(['Activity', 'create', {
-          case_id: item.id,
-          target_contact_id: role.contact_id,
-          status_id: 'Completed',
-          activity_type_id: role.relationship_type_id ? 'Remove Case Role' : 'Remove Client From Case',
-          subject: ts('%1 removed as %2', { 1: role.display_name, 2: role.role })
-        }]);
+          makeAPICalltoUnassignRole(apiCalls, 'Remove Client From Case', role);
+        });
+      } else {
+        promptForContact(
+          {
+            title: ts('Remove %1', { 1: role.role }),
```

Comment:
Don't we need still need the  message Remove %1 as XXX to be displayed on the modal?

Reply:
Added that to the model title now.